lib/rubocop/cop/rails/timeout_from_transaction.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for transactions wrapped in timeouts,
+      # motivated by change in Rails behavior that will
+      # cause exited transactions to be rolled back
+      # rather than committed (pre ActiveRecord 7 behavior).
+      #
+      # @example
+      #   # bad
+      #   Timeout.timeout(10) do
+      #     ApplicationRecord.transaction do
+      #       user.save
+      #     end
+      #   end
```

Comment:
What do you think of the example below?

```suggestion
      #   # bad
      #   Timeout.timeout(10) do
      #     ApplicationRecord.transaction do
      #       method_call_that_may_timeout
      #       user.save
      #     end
      #   end
      #
      #   # good
      #   Timeout.timeout(10) do
      #     method_call_that_may_timeout
      #   end
      #   ApplicationRecord.transaction do
      #     user.save
      #   end
```